crates/router/src/core/webhooks.rs

```diff
@@ -187,6 +188,165 @@ async fn refunds_incoming_webhook_flow<W: api::OutgoingWebhookType>(
+                    .into_report()
+                    .change_context(errors::WebhooksFlowError::DisputeCoreFailed)?,
+                payment_id: payment_id.to_owned(),
+                merchant_id: merchant_id.to_owned(),
+                connector_status: dispute_details.connector_status,
+                connector_dispute_id: dispute_details.connector_dispute_id,
+                connector_reason: dispute_details.connector_reason,
+                connector_reason_code: dispute_details.connector_reason_code,
+                challenge_required_by: dispute_details.challenge_required_by,
+                dispute_created_at: dispute_details.created_at,
+                updated_at: dispute_details.updated_at,
+            };
+            state
+                .store
+                .insert_dispute(new_dispute.clone())
+                .await
+                .change_context(errors::WebhooksFlowError::WebhookEventCreationFailed)
+        }
+        Some(dispute) => {
+            logger::info!("Dispute Already exists, Updating the dispute details");
+            let update_dispute = storage_models::dispute::DisputeUpdate::Update {
+                dispute_stage: dispute_details.dispute_stage.foreign_into(),
+                dispute_status: event_type
+                    .foreign_try_into()
+                    .into_report()
+                    .change_context(errors::WebhooksFlowError::DisputeCoreFailed)?,
+                connector_status: dispute_details.connector_status,
+                connector_reason: dispute_details.connector_reason,
+                connector_reason_code: dispute_details.connector_reason_code,
+                challenge_required_by: dispute_details.challenge_required_by,
+                updated_at: dispute_details.updated_at,
+            };
+            db.update_dispute(dispute, update_dispute)
+                .await
+                .change_context(errors::WebhooksFlowError::ResourceNotFound)
+        }
+    }
+}
+
+#[instrument(skip_all)]
+async fn disputes_incoming_webhook_flow<W: api::OutgoingWebhookType>(
+    state: AppState,
+    merchant_account: storage::MerchantAccount,
+    webhook_details: api::IncomingWebhookDetails,
+    source_verified: bool,
+    connector: &(dyn api::Connector + Sync),
+    request_details: &IncomingWebhookRequestDetails<'_>,
+    event_type: api_models::webhooks::IncomingWebhookEvent,
+) -> CustomResult<(), errors::WebhooksFlowError> {
+    if source_verified {
+        let db = &*state.store;
+        let dispute_details = connector
+            .get_dispute_details(request_details)
+            .change_context(errors::WebhooksFlowError::WebhookEventObjectCreationFailed)?;
+        let payment_attempt = get_payment_attempt_from_object_reference_id(
+            state.clone(),
+            webhook_details.object_reference_id,
+            &merchant_account,
+        )
+        .await?;
+        let option_dispute = db
+            .find_by_payment_id_connector_dispute_id(
```

Comment:
When we create the dispute in our system? only through `webhooks`?

Reply:
Yes, only throgh webhooks